RevenueCatUI/Templates/V2/Variables/VariableHandlerV2.swift

```diff
@@ -702,6 +746,70 @@ private extension VariablesV2 {
         }
     }
 
+    func productGrantAmount(package: Package, virtualCurrencyCode: String) -> String {
+        guard let grant = package.virtualCurrencyGrants[virtualCurrencyCode] else {
+            return ""
+        }
+
+        return "\(grant.amount)"
+    }
+
+    func productGrantName(package: Package,
+                          virtualCurrencyCode: String,
+                          virtualCurrencies: [String: VirtualCurrencyMetadata]) -> String {
+        guard let name = virtualCurrencies[virtualCurrencyCode]?.name else {
+            return ""
+        }
+
+        return name
+    }
+
+    func productGrantAllAmountCodePerPeriod(package: Package, localizations: [String: String]) -> String {
+        let periodAbbreviated = self.productPeriod(package: package, localizations: localizations)
+
+        var currencies: [String] = []
+        package.virtualCurrencyGrants.forEach { key, value in
+            currencies.append("\(value.amount) \(key)/\(periodAbbreviated)")
+        }
+        return currencies.joined(separator: " ")
```

Comment:
Instead of using `currencies.joined()`, would it be worth using [ListFormatter.localizedString()](https://developer.apple.com/documentation/foundation/listformatter) to try to get a localized list string here?

That way, instead of having "... ... ...", you could have "..., ..., and ..." in English, and the appropriate localized list for each other locale.